### DISCUSSION TITLE
Cannot use loading delay with display property

### DISCUSSION THREAD

**Discussion by godismyjudge95** [livewire/livewire#3004] [7. Bugs (archived)] [answered]
### Description
In a recent project I have come across an issue where you cannot use both the loading delay and display property.
I am unsure if this affects more of the loading directive properties or not.

It appears to be caused by the fact that livewire specifically targets styling like so:
```
[wire\:loading],
[wire\:loading\.delay],
[wire\:loading\.inline-block],
[wire\:loading\.inline],
[wire\:loading\.block],
[wire\:loading\.flex],
[wire\:loading\.table],
[wire\:loading\.grid]
```

Which means the loading directive only hides elements that specifically match those lines.  As an alternative to listing out every possible combination of directives and properties, I would recommend a simple javascript solution to target any element with an attribute beginning with `wire:loading` and not containing the `.remove` property.

### Exact steps to reproduce
1. Create a livewire component (might want to insert a longer operation in there so the loading action is performed)
2. In the livewire component's view file paste both of the snippets below
3. Load a page with the component
4. Notice the `wire:loading.block.delay` one is not hidden after the loading is done.

### Stripped-down, copy-pastable code snippets
Works:
```
<div wire:loading.delay>Loading</div>
```
Does not work:
```
<div wire:loading.block.delay>Loading</div>
```

**Context**
- Livewire version: 2.4.4
- Laravel version: 8.41.0
- Alpine version: 2.8.2
- Browser: Chrome v90

**Comment by pyr0hu** [comment]
Any solution @godismyjudge95 ? The issue got closed without any answer and this discussion seems dead.

**Reply by joshhanley** [reply]
@pyr0hu due to css selectors, we can't chain multiple things on to `wire:loading`. One thing you could try though is putting two calls on the component like

```blade
<div wire:loading.block wire:loading.delay>Loading</div>
```

Hope this helps!

**Reply by godismyjudge95** [reply]
@pyr0hu yeah, sorry I didn't come back and update this.  @joshhanley has the correct answer - although I can't mark replies as answers for this discussion...

**Reply by joshhanley** [reply]
@godismyjudge95 have duplicated it below and marked as answered 🙂 but yeah I wish you could mark replies as answers too

**Comment by joshhanley** [comment, marked as answer]
Due to css selectors, we can't chain multiple things on to wire:loading. One thing you could try though is putting two calls on the component like

`<div wire:loading.block wire:loading.delay>Loading</div>`
Hope this helps!

**Reply by Iheanyi001** [reply]
this only works for wire:loading.block and doesn't seems to work when using wire:loading.flex

**Reply by joshhanley** [reply]
@Iheanyi001 yeah that whole system is quite limited, due the css selectors as described above.

I would suggest not using flex, and wrapping your div that needs flex, in another div, and putting wire:loading on that instead, so it can be displayed as a block, which enables delay to work as expected too.

Hope this helps!

**Comment by loranger** [comment]
Hi,

I'm afraid this solutions does not work anymore:

I use livewire 2.10 and when I use both `wire:loading.delay` and `wire:loading.flex`, the delay does not apply…

Do you have a workaround ?

**Reply by samfrjn11** [reply]
I found a working work-around:

Just nest 2 divs, for example like so:
```
 <div class="absolute z-40 inset-x-0" wire:loading.block wire:target="keyword">
            <div wire:loading.delay.longest class="w-full">
                <div class="animate-pulse p-4">
                    <div class="py-1">
                        <div class="h-3 bg-gray-300 rounded w-5/6"></div>
                        <div class="mt-4 w-1/3">
                            <div class="grid grid-cols-3 gap-2">
                                <div class="h-3 bg-gray-300 rounded col-span-1"></div>
                                <div class="h-3 bg-gray-300 rounded col-span-2"></div>
                            </div>
                        </div>
                    </div>
                </div>
            </div>
        </div>
```

**Reply by SoNiice** [reply]
> I found a working work-around:
> 
> Just nest 2 divs, for example like so:
> 
> ```
>  <div class="absolute z-40 inset-x-0" wire:loading.block wire:target="keyword">
>             <div wire:loading.delay.longest class="w-full">
>                 <div class="animate-pulse p-4">
>                     <div class="py-1">
>                         <div class="h-3 bg-gray-300 rounded w-5/6"></div>
>                         <div class="mt-4 w-1/3">
>                             <div class="grid grid-cols-3 gap-2">
>                                 <div class="h-3 bg-gray-300 rounded col-span-1"></div>
>                                 <div class="h-3 bg-gray-300 rounded col-span-2"></div>
>                             </div>
>                         </div>
>                     </div>
>                 </div>
>             </div>
>         </div>
> ```

This sadly doesn't work in my case since it's still applying inline-block and that's ruining my layout

**Reply by samfrjn11** [reply]
What if you run an inline style `display: block !important;`?

**Reply by SoNiice** [reply]
Doesn't help in my case, tried on every element to be sure if it helps in any way, it sadly didn't
